File: app/src/main/java/com/jeronimo/swingradio/BlogFragment.java
package com.jeronimo.swingradio;

import android.app.Fragment;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.KeyEvent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebChromeClient;
import android.webkit.WebView;
import android.webkit.WebViewClient;

/**
 * Created by gleb on 23.12.15.
 */
public class BlogFragment extends Fragment {

    WebView webView;
    private String url;
    private String host;
    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
    {

        View view = inflater.inflate(R.layout.fragment_blog, container, false);
        url = getResources().getString(R.string.SR_BLOG_URL);
        host = getResources().getString(R.string.SR_BLOG_HOST);
        webView = (WebView)view.findViewById(R.id.fragment_blog_webView);
        WebViewClient webViewClient = new MyWebViewClient();
        WebChromeClient webChromeClient = new WebChromeClient();
        webView.setWebViewClient(webViewClient);
        webView.setWebChromeClient(webChromeClient);
        webView.loadUrl(url);
        webView.setOnKeyListener(new View.OnKeyListener() {
            @Override
            public boolean onKey(View v, int keyCode, KeyEvent event) {
                if (event.getAction() == KeyEvent.ACTION_DOWN) {
                    WebView webView = (WebView) v;

                    switch (keyCode) {
                        case KeyEvent.KEYCODE_BACK:
                            if (webView.canGoBack()) {
                                webView.goBack();
                                return true;
                            }
                            break;
                    }
                }

                return false;
            }
        });
        return view;
    }

    public class MyWebViewClient extends WebViewClient {

        @Override
        public boolean shouldOverrideUrlLoading(WebView view, String url) {
            if(Uri.parse(url).getHost().contains(host)) {
                return false;
            }

            Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
            view.getContext().startActivity(intent);
            return true;
        }
    }
}

File: app/src/main/java/com/jeronimo/swingradio/BlogFragment.kt
package com.jeronimo.swingradio

import android.app.Fragment
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.KeyEvent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.webkit.WebChromeClient
import android.webkit.WebView
import android.webkit.WebViewClient
import kotlinx.android.synthetic.main.fragment_blog.*

/**
 * Created by gleb on 23.12.15.
 */
class BlogFragment : Fragment() {

    private var url: String? = null
    private var host: String? = null
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_blog, container, false)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        url = resources.getString(R.string.SR_BLOG_URL)
        host = resources.getString(R.string.SR_BLOG_HOST)
        val webViewClient = MyWebViewClient()
        val webChromeClient = WebChromeClient()
        fragment_blog_webView.setWebViewClient(webViewClient)
        fragment_blog_webView.setWebChromeClient(webChromeClient)
        fragment_blog_webView.loadUrl(url)
        fragment_blog_webView.setOnKeyListener(View.OnKeyListener { v, keyCode, event ->
            if (event.action == KeyEvent.ACTION_DOWN) {
                val webView = v as WebView

                when (keyCode) {
                    KeyEvent.KEYCODE_BACK -> if (webView.canGoBack()) {
                        webView.goBack()
                        return@OnKeyListener true
                    }
                }
            }

            false
        })
    }

    inner class MyWebViewClient : WebViewClient() {

        override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
            if (Uri.parse(url).host.contains(host!!)) {
                return false
            }

            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
            view.context.startActivity(intent)
            return true
        }
    }
}
